test/controls/createControls.js

```diff
@@ -0,0 +1,46 @@
+import jsdom from 'jsdom';
+import createControls from '../../src/createControls';
+import expect from 'expect';
+
+describe('controls object creation', () => {
+    const { JSDOM } = jsdom;
+    let dom,
+        container,
+        controls;
+
+    before(() => {
+        dom = new JSDOM('<!DOCTYPE html>');
+        container = dom.window.document.createElement('div');
+    });
+
+    beforeEach(() => {
+        controls = createControls(container);
+    });
+
+    afterEach(() => {
+        container.children[0].remove();
+        controls = null;
+    });
+
```

Comment:
Is there any data mapping checking going on in createControls for type="subsetter"?